core/tests/puppeteer-acceptance-tests/spec/exploration-creator-tests/create-exploration-and-change-basic-settings.spec.ts

```diff
@@ -0,0 +1,125 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Acceptance Test for Exploration Creator and Exploration Manager
+ */
+
+import testConstants from '../../puppeteer-testing-utilities/test-constants';
+import {UserFactory} from '../../puppeteer-testing-utilities/user-factory';
+import {ExplorationCreator} from '../../user-utilities/exploration-creator-utils';
+import {SuperAdmin} from '../../user-utilities/super-admin-utils';
+
+const DEFAULT_SPEC_TIMEOUT = testConstants.DEFAULT_SPEC_TIMEOUT;
+
+describe('Exploration Creator', function () {
+  let explorationCreator: ExplorationCreator;
+  let explorationVisitor: ExplorationCreator;
+  let superAdmin: SuperAdmin;
+
+  beforeAll(async function () {
+    explorationCreator = await UserFactory.createNewUser(
+      'explorationAdm',
+      'exploration_creator@example.com'
+    );
+    explorationVisitor = await UserFactory.createNewUser(
+      'explorationSeeker',
+      'exploration_visitor@example.com'
+    );
+    superAdmin = await UserFactory.createNewSuperAdmin('Leader');
+    const guestUser1 = await UserFactory.createNewUser(
+      'guestUsr1',
+      'guest_user1@example.com'
+    );
+    await guestUser1.closeBrowser();
+
+    const guestUser2 = await UserFactory.createNewUser(
+      'guestUsr2',
+      'guest_user2@example.com'
+    );
+    await guestUser2.closeBrowser();
+
+    const guestUser3 = await UserFactory.createNewUser(
+      'guestUsr3',
+      'guest_user3@example.com'
+    );
+    await guestUser3.closeBrowser();
+
+    await superAdmin.assignRoleToUser('explorationAdm', 'voiceover admin');
+    await superAdmin.expectUserToHaveRole('explorationAdm', 'voiceover admin');
+
+    await superAdmin.assignRoleToUser('explorationAdm', 'curriculum admin');
+    await superAdmin.expectUserToHaveRole('explorationAdm', 'curriculum admin');
+
+    await superAdmin.closeBrowser();
+  }, DEFAULT_SPEC_TIMEOUT);
+
+  it(
+    'should create an exploration and modify it via the Settings tab',
+    async function () {
+      await explorationCreator.openCreatorDashboardPage();
+      await explorationCreator.switchToEditorTab();
+      await explorationCreator.updateExplorationIntroText(
+        'Exploration intro text'
+      );
+      await explorationCreator.updateCardName('Test');
+      await explorationCreator.addEndInteraction();
+
+      await explorationCreator.goToSettingsTab();
+
+      await explorationCreator.addTitle('Your Title Here');
+      await explorationCreator.expectTitleToHaveMaxLength(36);
```

Comment:
This method doesn't make sense. The title you're giving already has a length of < 36 so I'm not sure what system behaviour this is testing.

If you want to check that bad titles aren't accepted then, as the user, you'd need to try entering a title that's too long and then observe the result (.g. that it's cut off).

Reply:
@seanlip If we enter long title then tests will fail as per the implementation -> 
``` typescript
async expectTitleToHaveMaxLength(maxLength: number): Promise<void> {
    const titleInput = await this.page.$('.e2e-test-exploration-title-input');
    const title = await this.page.evaluate(input => input.value, titleInput);
    const titleLength = title.length;

    if (titleLength <= maxLength) {
      showMessage(
        'Title length is within the' +
          ` allowed limit of ${maxLength} characters.`
      );
    } else {
      throw new Error(
        'Title length exceeds the allowed' +
          ` limit of ${maxLength} characters.`
      );
    }
  }
```

due to -> 
```typescript
throw new Error(
        'Title length exceeds the allowed' +
          ` limit of ${maxLength} characters.`
)
```

PTAL.

Reply:
I think there might be a misunderstanding here. What your method is doing is checking that the title seen on the page is less than a length specified in the code. What the acceptance tests should be doing, in general, is mimicking a user's behaviour and verifying that they "see what they expect to see".

Modifying your method as described doesn't fix this underlying issue. What happens in practice when the creator tries to enter a title that's too long? Have the tests play through that scenario.